src/Client/GuzzleFileFetcher.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Tuf\Client;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Tuf\Exception\DownloadSizeException;
+use Tuf\Exception\RepoFileNotFound;
+use Tuf\JsonNormalizer;
+
+/**
+ * Defines a file fetcher that uses Guzzle to read a file over HTTPS.
+ */
+class GuzzleFileFetcher implements RepoFileFetcherInterface
+{
+    /**
+     * The HTTP client.
+     *
+     * @var \GuzzleHttp\ClientInterface
+     */
+    private $client;
+
+    /**
+     * GuzzleFileFetcher constructor.
+     *
+     * @param \GuzzleHttp\ClientInterface $client
+     *   The HTTP client.
+     */
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Creates an instance of this class with a specific base URI.
+     *
+     * @param string $baseUri
+     *   The base URI from which to fetch files.
+     *
+     * @return static
+     *   A new instance of this class.
+     */
+    public static function createFromUri(string $baseUri) : self
```

Comment:
Is there a reason we need this to be public except for the tests? Won't we expect most cases to use `createFromUri()`?  Or do we want to support the use of passing in something other than `\GuzzleHttp\Client`?

Reply:
`createFromUri()` is the actual entry point I expect calling code to use. The only reason the constructor is public is to facilitate unit testing.

Reply:
I think it would nice if we could make the constructor private if we don't see it as an entry point. here is an idea on that https://github.com/tedbow/php-tuf/pull/1/commits/89df373b2da4b68cc68b8bdf26848aa5e5264c7d#diff-22262933f442133c3659f5b829b1cb3cd89362e42f52b48474c514eacfa99aaaR71

Reply:
I'd prefer not to do that, if I can, because it's "dishonest" -- it also couples the test to the _internal_ implementation of the class, rather than strictly testing what it _does_. The pattern I already have is basically plain dependency injection. So...are you sure? :)

Reply:
It's fine